### DISCUSSION TITLE
How to implement WSSE authentication in Symfony 5.4 and beyond ?

### DISCUSSION THREAD

**Discussion by sdespont** [symfony/symfony#47850] [Q&A] [answered]
3 years ago, I have implemented a WSSE authentication method based on [this Symfony user guide](https://symfony.com/doc/5.2/security/custom_authentication_provider.html#meet-wsse). 

After upgrading to version 5.4, a lot of Security classes have been deprecated.

Unfortunately, there is no longer documentation about WSSE in the 5.4 documentation 😟

Because there is [no authenticator suggested by Symfony](https://symfony.com/doc/5.4/security.html#security-authenticators) doc about WSSE, how can I migrate to the new security system ? Do I need to create a [custom authenticator](https://symfony.com/doc/5.4/security/custom_authenticator.html) ?

Any advice would be very appreciate about WSSE authentication migration 🙏

**Comment by stof** [comment, marked as answer]
Yes, you need to create a custom authenticator, as WSSE is not in core (btw, in your old code, you were also creating a custom authentication method as WSSE was not implemented in core either)

**Reply by sdespont** [reply]
Thank you for your answer. You are right, the previous authenticator was a custom one, but created following the Symfony documentation which explaining the implementation step by step. Do you confirm that there is no available exemple of this kind of custom authenticator?

**Reply by stof** [reply]
IIRC, we have examples in the doc. But maybe not using WSSE.

**Comment by sdespont** [comment]
Here is my [custom authenticator for WSSE](https://gist.github.com/sdespont/d4bb9538e3bf6076627127bde8afa7d7)

Perhaps it will help someone.
